go/stream.go

```diff
@@ -297,12 +303,16 @@ func (s *stream) Close() (err error) {
 	if !s.closed.CompareAndSwap(false, true) {
 		return nil
 	}
-	defer close(s.output)
+	s.streamCtxCancel()
+	// this lock ensures websocket readers stop in a safe spot for closing
+	s.closingMutex.Lock()
+	defer s.closingMutex.Unlock()
 
+	// mutex here to end of function
```

Comment:
This comment seems like it may be in the wrong spot?

Reply:
removed